Results_Engine/Query/DisplayObjectResult.cs

```diff
@@ -0,0 +1,150 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using BH.oM.Base;
+using System.Linq;
+using System.Reflection;
+using BH.oM.Geometry;
+using BH.oM.Structure.Results;
+using BH.oM.Graphics;
+using System;
+using BH.Engine.Graphics;
+using BH.Engine.Geometry;
+using BH.Engine.Base;
+using BH.Engine.Analytical;
+using BH.oM.Analytical.Results;
+using BH.oM.Analytical.Elements;
+using BH.oM.Graphics.Colours;
+using BH.oM.Quantities.Attributes;
+
+namespace BH.Engine.Results
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Generates a list of coloured geometry based on provided BHoMObejcts and ObjectResults.")]
+        [Input("objects", "BHoMObjects to colour.")]
+        [Input("results", "The IObjectResults to colour by.")]
+        [Input("objectIdentifier", "Should either be a string specifying what property on the object that should be used to map the objects to the results, or a type of IAdapterId fragment to be used to extract the object identification, i.e. which fragment type to look for to find the identifier of the object. If no identifier is provided, the object will be scanned an IAdapterId to be used.")]
+        [Input("caseFilter", "Which cases to colour by, default is all.")]
+        [Input("displayProperty", "The name of the property on the result to colour by. If nothing is provided, the first available property will be used.")]
+        [Input("gradientOptions", "How to color the mesh, null defaults to `BlueToRed` with automatic range.")]
+        [MultiOutput(0, "results", "A List of Lists of RenderGeometry, where the outer list corresponds to the object and the inner list correspond to the matchis results..")]
+        [MultiOutput(1, "gradientOptions", "The gradientOptions that were used to colour the meshes.")]
+        public static Output<List<List<RenderGeometry>>, GradientOptions> DisplayObjectResult(this IEnumerable<IBHoMObject> objects, IEnumerable<IObjectResult> results, string displayProperty = "", object objectIdentifier = null, List<string> caseFilter = null, GradientOptions gradientOptions = null)
```

Comment:
To allow for a CaseFilter input, shouldn't we place a type constraint on the `results` input to be `ICasedResults`?

I would probably have a separate overloaded method to allow for that CaseFilter input/result type combination

Reply:
Agree this needs to be expanded. Have some thinking around this, and have raised an issue here to capture my initial thoughts: https://github.com/BHoM/BHoM_Engine/issues/2772

This fundamentally comes down to a further expansion of the capabilities of MapResults filtering capabilities, and for sure is something that should be moved further. My initial plan was to do this in a later PR, but if you think it is better done for this PR, I can add the functionality.

Reply:
Actually, after another think, I think this might be the PR to make this change, to avoid having to change the interface of the method again. Will get a first version in, that might not be perfect and cover all cases yet, but work as a start that can be expanded upon in a more isolated fashion.